src/main/java/battlestates/states/EnemyTurnState.java

```diff
@@ -0,0 +1,57 @@
+package battlestates.states;
+
+import interfaces.State;
+import io.InputValidator;
+import objects.battle.StatDisplayer;
+import objects.battle.enemy.ai.EnemyActionHandler;
+import objects.character.EnemyFacade;
+import objects.character.Player;
+
+public class EnemyTurnState implements State {
+    private Player user;
+    private EnemyFacade foe;
+    private boolean done = false;
+    private String userAction;
+
+    public EnemyTurnState(Player user, EnemyFacade foe, String userAction) {
+        this.user = user;
+        this.foe = foe;
+        this.userAction = userAction;
+    }
+    @Override
+    public void preInput() {
+        EnemyActionHandler action = foe.enemyAction(userAction);
```

Comment:
I implemented a useGimmick() for enemy as well so make sure you call useGimmick() and you will not have to cast anymore since it will return interface and without knowing the type of the enemy, you can use all the methods you want.